src/main/java/sirius/biz/vfs/ftp/FTPBridge.java

```diff
@@ -131,14 +134,18 @@ private void setupNetwork(ListenerFactory factory) {
             factory.setServerAddress(bindAddress);
         }
 
-        if (Strings.isFilled(passivePorts)) {
-            DataConnectionConfigurationFactory dataConnectionConfigurationFactory =
-                    new DataConnectionConfigurationFactory();
+        DataConnectionConfigurationFactory dataConnectionConfigurationFactory =
+                new DataConnectionConfigurationFactory();
 
+        if (Strings.isFilled(passivePorts)) {
             dataConnectionConfigurationFactory.setPassivePorts(passivePorts);
+        }
 
-            factory.setDataConnectionConfiguration(dataConnectionConfigurationFactory.createDataConnectionConfiguration());
+        if (Strings.isFilled(passiveExternalAddress)) {
+            dataConnectionConfigurationFactory.setPassiveExternalAddress(passiveExternalAddress);
         }
+
+        factory.setDataConnectionConfiguration(dataConnectionConfigurationFactory.createDataConnectionConfiguration());
```

Comment:
is this a bug or why is this called even if passivePorts isEmpty?

Reply:
Because passiveExternalAddress could've been set too. Calling this without setting passivePorts nor passiveExternalAddress has the same effect as not calling it. It will simply use the defaults.